src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -188,73 +205,61 @@ await signInManager.ValidateSecurityStampAsync(refreshTicket.Principal) is not T
             return TypedResults.Ok();
         });
 
-        routeGroup.MapPost("/resetPassword", async Task<Results<Ok, ValidationProblem>>
-            ([FromBody] ResetPasswordRequest resetRequest, [FromServices] IServiceProvider sp) =>
+        routeGroup.MapPost("/forgotPassword", async Task<Results<Ok, ValidationProblem>>
+            ([FromBody] ForgotPasswordRequest resetRequest, [FromServices] IServiceProvider sp) =>
         {
             var userManager = sp.GetRequiredService<UserManager<TUser>>();
+            var user = await userManager.FindByEmailAsync(resetRequest.Email);
 
-            if (!string.IsNullOrEmpty(resetRequest.ResetCode) && string.IsNullOrEmpty(resetRequest.NewPassword))
+            if (user is not null && await userManager.IsEmailConfirmedAsync(user))
             {
-                return CreateValidationProblem("MissingNewPassword", "A password reset code was provided without a new password.");
+                var code = await userManager.GeneratePasswordResetTokenAsync(user);
+                code = WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(code));
+
+                await emailSender.SendEmailAsync(resetRequest.Email, "Reset your password",
+                    $"Reset your password using the following code: {HtmlEncoder.Default.Encode(code)}");
             }
 
+            // Don't reveal that the user does not exist or is not confirmed, so don't return a 200 if we would have
```

Comment:
Actually, there aren't any logs in this file. I guess it's somewhat by design so we don't leak PII?

Reply:
SignInManager and UserManager do a fair amount of logging themselves. That's generally a better layer for logging because then Identity UI gets the logging as well.